### PR TITLE
Allow unicode keys

### PR DESCRIPTION
**Pull request by qoelet**
Fixes DOR-785

### CODE REVIEW

**Comment by qoelet** [comment]
@penmanglewood @cdenis95

File: memcache/memcache.go [hunk 1]
```diff
@@ -89,7 +89,7 @@ func legalKey(key string) bool {
 		return false
 	}
 	for i := 0; i < len(key); i++ {
-		if key[i] <= ' ' || key[i] > 0x7e {
+		if key[i] <= ' ' || key[i] == 0x7f {
```
Comment (cdenis95) [inline memcache/memcache.go:92]: What's the logic here? Why exclude 0x7f specifically?

Comment (qoelet) [reply memcache/memcache.go:92]: https://github.com/memcached/memcached/blob/master/doc/protocol.txt#L41
Control character.

Comment (cdenis95) [reply memcache/memcache.go:92]: But, there are plenty more control characters, other than 0x7f

Comment (qoelet) [reply memcache/memcache.go:92]: In ASCII they are below 27, which is covered by the first comparison clause. After that, only 127 remains.

Comment (qoelet) [reply memcache/memcache.go:92]: (and unless I'm missing something, this entire malformed key checking only extends to the ASCII space, i.e. it shouldn't matter for say UTF-8)

Comment (cdenis95) [reply memcache/memcache.go:92]: The whole point is for us to support non ASCII characters,for ventures that use non-latin letters, isn't it? 
For ex, thai characters: (0E00 - 0E7F) http://www.unicode.org/charts/PDF/U0E00.pdf

**Pull request merged**